### PR TITLE
Improve performance of JSON import and add progress bar

### LINKED ISSUE #7612: Import from JSON is extremely slow with large imports
#### Summary
Importing very large JSON files (on the order of megabytes) is extremely slow. This would be less of an issue if Open MCT gave the user some feedback that it was doing something, but it doesn't.

#### Expected vs Current Behavior
Import from JSON has some inefficiencies which could be improved to speed up the import process. The user should be informed that an import is in progress.

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [x] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by scottbell**
Closes #7612 

### Describe your changes:

Instead of using `JSON.parse`, `JSON.stringify`, and Regex, walk the tree instead. This speeds up the import considerably. I also added a progress bar so the user can see the status of the import.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

**Comment by scottbell** [comment]
With this export ([Big Test Folder (2).json.zip](https://github.com/nasa/openmct/files/14805785/Big.Test.Folder.2.json.zip)), the speed goes from > 5 minutes, to a few seconds:

https://github.com/nasa/openmct/assets/9853862/966c4304-f136-4a2d-abdf-75621a6b4e25

File: src/plugins/importFromJSONAction/ImportFromJSONActionSpec.js [hunk 1]
```diff
@@ -123,8 +122,6 @@ describe('The import JSON action', function () {
         Object.prototype.hasOwnProperty.call(newObject, '__proto__') ||
         Object.prototype.hasOwnProperty.call(Object.getPrototypeOf(newObject), 'toString');
 
-      // warning from openmct.objects.get
-      expect(console.warn).not.toHaveBeenCalled();
```
Comment (scottbell) [inline src/plugins/importFromJSONAction/ImportFromJSONActionSpec.js:127]: vue warnings were causing tests to fail here

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js [hunk 2]
```diff
@@ -170,9 +276,16 @@ export default class ImportAsJSONAction {
    * @param {Object} objTree
    */
   async _importObjectTree(domainObject, objTree) {
```
Comment (ozyx) [inline src/plugins/importFromJSONAction/ImportFromJSONAction.js:278]: can we switch these to actual private methods without ruining the unit tests?

**Review by ozyx — CHANGES_REQUESTED** [review]
Nice work, this is awesome! I've got some comments-- nothing blocking but some readability improvements and other nitpicks before we merge

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js (hunk 2 above)
Comment (scottbell) [reply src/plugins/importFromJSONAction/ImportFromJSONAction.js:280]: Changing these to private methods cause issues with the [way we're returning](https://github.com/nasa/openmct/blob/41d0b5acce3f99c0f8086d2b19bd1dbe4360b640/src/plugins/importFromJSONAction/ImportFromJSONAction.js#L376) from `_showForm`. E.g., changing to `#importObjectTree` causes:
```
ImportFromJSONAction.js:77 Uncaught (in promise) TypeError: Receiver must be an instance of class ImportAsJSONAction
    at ImportAsJSONAction.onSave (ImportFromJSONAction.js:77:1)
    at eval (ImportFromJSONAction.js:376:1)
```

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**